### PR TITLE
[Compiler Tools] add some tools to wrtite instructions.

### PR DESCRIPTION
**Pull request by Jim8y**
This pr add some tool methods that concludes the instruction sets. 

These tool methods can not only help us to write instructions, but also help us to optimize the price (with cheaper way of achieving) in the future.

### CODE REVIEW

File: src/Neo.Compiler.CSharp/MethodConvert/MethodConvert.cs [hunk 1]
```diff
@@ -667,13 +667,99 @@ private void CreateObject(SemanticModel model, ITypeSymbol type, InitializerExpr
...
 
+        /// <summary>
+        /// Duplicate an item in the stack without change anything else
+        ///  For example, dup(2)
+        ///     a - b - c => a - b - b - c.
+        /// </summary>
+        /// <param name="index">Index of items to be duped.</param>
+        private void DupAt(int index)
+        {
+            if (index <= 0) throw new Exception("Invalid index for Dup");
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert/MethodConvert.cs:688]: Switch?

File: src/Neo.Compiler.CSharp/MethodConvert/MethodConvert.cs [hunk 2]
```diff
@@ -667,13 +667,99 @@ private void CreateObject(SemanticModel model, ITypeSymbol type, InitializerExpr
...
+
+        /// <summary>
+        /// Drop items at index n from the stack.
+        /// </summary>
+        /// <param name="index">Index of the items to be dropped.</param>
+        /// <exception cref="Exception">When index is invalid.</exception>
+        private void DropAt(int index)
+        {
+            if (index <= 0) throw new Exception("Invalid index for DropAt");
+            if (index == 1)
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert/MethodConvert.cs:732]: Why index doesn't start in 0?

File: tests/Neo.Compiler.CSharp.UnitTests/UnitTest_String.cs [hunk 3]
```diff
@@ -100,10 +100,10 @@ public void Test_TestEndWith()
         public void Test_TestContains()
         {
             Assert.IsTrue(Contract.TestContains("hello world"));
-            Assert.AreEqual(2032920, Engine.FeeConsumed.Value);
+            Assert.AreEqual(2032800, Engine.FeeConsumed.Value);
```
Comment (shargon) [inline tests/Neo.Compiler.CSharp.UnitTests/UnitTest_String.cs:103]: Why changed the gas in the methods are not used?

Comment (Hecate2) [reply tests/Neo.Compiler.CSharp.UnitTests/UnitTest_String.cs:103]: > Why changed the gas in the methods are not used?

I do not really know what happened, but we do not have `TestInsert` method in the master branch, and we have an artifact Contract_String that does not match the test contract source codes

File: src/Neo.Compiler.CSharp/MethodConvert/MethodConvert.cs (hunk 1 above)
Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert/MethodConvert.cs:688]: these does not mater at all, can be this way, can be that way. Wont affect what this pr intents for.

**Pull request closed without merging**